sched/scheduler/stateful_scheduler.go

```diff
@@ -1016,6 +1017,30 @@ func (s *statefulScheduler) GetSagaCoord() saga.SagaCoordinator {
 	return s.sagaCoord
 }
 
+func (s *statefulScheduler) OfflineWorker(req sched.OfflineWorkerReq) error {
+	if !stringInSlice(req.Requestor, s.config.Admins) {
+		return fmt.Errorf("Requestor %s unauthorized to offline worker", req.Requestor)
+	}
+	log.Infof("Offlining worker %s", req.ID)
+	n := cluster.NodeId(req.ID)
+	s.clusterState.updateCh <- []cluster.NodeUpdate{cluster.NewUserInitiatedRemove(n)}
+	return nil
+}
+
+func (s *statefulScheduler) ReinstateWorker(req sched.ReinstateWorkerReq) error {
+	if !stringInSlice(req.Requestor, s.config.Admins) {
+		return fmt.Errorf("Requestor %s unauthorized to reinstate worker", req.Requestor)
+	}
+	n := cluster.NodeId(req.ID)
+	if ns, ok := s.clusterState.offlinedNodes[n]; !ok {
```

Comment:
Worth checking here? No check in Offline request, and if for some reason the state changes between now and when the request is processed, the requestor won't know what happened anyway.

Reply:
True - will remove.

Reply:
Ah - remembered we need this check, ns is used for the add. Can put a similar check in OfflineWorker for consistency.